### ISSUE TITLE
Refresh ESNIKeys with HelloRetryRequest

### ISSUE THREAD

**Issue by ghedo** [ekr/draft-rescorla-tls-esni#17]
Regarding:

```
- If the EncryptedSNI.label value does not correspond to any known
  SNI encryption key, it MUST abort the connection with an
  "illegal_parameter" alert.
  [[OPEN ISSUE: We looked at ignoring the extension but concluded
  this was better.]]
```
Rather than aborting, would it make more sense for the server to (optionally?) send a fresh `ESNIKeys` as an extension as part of `HelloRetryRequest`? Example use case would be that for whatever reason (e.g. key compromise?) the server might want to discard a previously advertised `ESNIKeys` before it expires. Not sure if it's worth the added complexity though.

(With some other changes it could also be used as an alternative way to publish a key at the cost of one round trip, but the usefulness there is probably limited)

**Comment by ekr** [comment]
I thought about this a bit, but making sure it doesn't allow cut-and-paste attacks seemed like it was going to need a lot of thought.

**Comment by chris-wood** [comment]
@ghedo I don't think we should do this. It allows an on-path TCP proxy to inject a HRR, with a fresh ESNI key, and cause the client to reveal its SNI to that proxy.
